flexeval/core/chat_dataset/base.py

```diff
@@ -79,5 +79,13 @@ def require_incremental_response(self) -> bool:
         """
         raise NotImplementedError
 
+    @abstractmethod
```

Comment:
ChatDataset has category_key function to represent category_key

Reply:
Not all `ChatDataset` has `category_key` so let's avoid have this method in the abstract method.
(The abstract class should only represent the core functions/attributes, which all on its concrete classes have)

Instead, how about specifying which key to look for at the ` LLMScore/ChatLLMScore` side?

Reply:
> Instead, how about specifying which key to look for only at the LLMScore/ChatLLMScore side, assuming such key is present in the extra_info filed of ChatInstance.

That sounds reasonable. I'll write code in this way.